apps/server/src/modules/authorization-rules/rules/school.rule.spec.ts

```diff
@@ -93,81 +80,353 @@ describe('SchoolRule', () => {
+						Permission.SCHOOL_VIEW,
+						Permission.CAN_EXECUTE_INSTANCE_OPERATIONS,
+					]);
 
-				const result = rule.hasPermission(user, school, context);
+				it('should be return true', () => {
+					const { user, school, context } = setup();
 
-				expect(result).toBe(true);
+					const result = rule.hasPermission(user, school, context);
+
+					expect(result).toBe(true);
+				});
 			});
-		});
 
-		describe('when user does not have required permissions', () => {
-			const setup = () => {
-				const { user, school } = setupSchoolAndUser();
-				const context = AuthorizationContextBuilder.read([]);
+			describe('when user has CAN_EXECUTE_INSTANCE_OPERATIONS and SCHOOL_EDIT', () => {
+				const setup = () =>
+					createParamsFromUserAndContextPermissions([
+						Permission.SCHOOL_EDIT,
+						Permission.CAN_EXECUTE_INSTANCE_OPERATIONS,
+					]);
+
+				it('should be return false', () => {
+					const { user, school, context } = setup();
+
+					const result = rule.hasPermission(user, school, context);
+
+					expect(result).toBe(false);
+				});
+			});
+
+			describe('when user has SCHOOL_VIEW', () => {
+				const setup = () => createParamsFromUserAndContextPermissions([Permission.SCHOOL_VIEW]);
+
+				it('should be return true', () => {
+					const { user, school, context } = setup();
+
+					const result = rule.hasPermission(user, school, context);
+
+					expect(result).toBe(true);
+				});
+			});
+
+			describe('when user has SCHOOL_EDIT', () => {
+				const setup = () => createParamsFromUserAndContextPermissions([Permission.SCHOOL_EDIT]);
+
+				it('should be return false', () => {
+					const { user, school, context } = setup();
+
+					const result = rule.hasPermission(user, school, context);
+
+					expect(result).toBe(false);
+				});
+			});
+
+			describe('when user has no required permission', () => {
+				const setup = () => createParamsFromUserAndContextPermissions([]);
+
+				it('should be return false', () => {
+					const { user, school, context } = setup();
+
+					const result = rule.hasPermission(user, school, context);
+
+					expect(result).toBe(false);
+				});
+			});
+
+			describe('when school is user school', () => {
+				const setup = () => createParamsFromUserAndContextPermissions([Permission.SCHOOL_VIEW]);
+
+				it('should be return true', () => {
+					const { user, school, context } = setup();
+
+					const result = rule.hasPermission(user, school, context);
+
+					expect(result).toBe(true);
+				});
+			});
+
+			describe('when school is user not school', () => {
+				const setup = () => {
+					const context = AuthorizationContextBuilder.read([]);
+					const school = schoolFactory.build();
+					const otherSchoolEntity = schoolEntityFactory.buildWithId();
+					const user = userFactory
+						.withPermissionsInRole([Permission.SCHOOL_VIEW])
+						.buildWithId({ school: otherSchoolEntity });
+
+					return { user, school, context };
+				};
+
+				it('should be return false', () => {
```

Comment:
Ich würde sagen, hier fehlt ein Test Case: Wenn der User nicht in der Schule ist, aber Instanz-Rechte hat.

Reply:
Hinzugefügt